### DISCUSSION TITLE
Does :core:testing module gets included in production release build?

### DISCUSSION THREAD

**Discussion by JackEblan** [android/nowinandroid#1254] [Q&A]
I also want to verify if the fakes also get included in production. Is it okay to include fake classes from testing to production (To minimize a lil bit of apk size🤐)

**Comment by SimonMarquis** [comment]
> Does :core:testing module gets included in production release build?

As long as it remain included only in "test" configurations like `testImplementation` or `androidTestImplementation`.
Which is the case at the moment: https://github.com/search?q=repo%3Aandroid%2Fnowinandroid+%22%28projects.core.testing%29%22&type=code


> Is it okay to include fake classes from testing to production

Generally speaking, it is not okay to include fake (test) classes in the production app, simply because it is not meant to be used in production.

**Reply by JackEblan** [reply]
Thank you for clarification @SimonMarquis My question is already late as the fakes already been moved to data-test😁

**Reply by JackEblan** [reply]
Can I ask about these classes here?
[projects.core.network](https://github.com/android/nowinandroid/blob/main/core/network/src/main/kotlin/com/google/samples/apps/nowinandroid/core/network/fake) as the fakes are still here.

**Reply by SimonMarquis** [reply]
You are right, `FakeNiaNetworkDataSource` is actually bundled in the release sourceSet.
That being said, it is actually used in the `demo` flavor:
https://github.com/android/nowinandroid/blob/ab9033421017962c5af6d7a1dd221997439210da/core/network/src/demo/kotlin/com/google/samples/apps/nowinandroid/core/network/di/FlavoredNetworkModule.kt#L31

And I guess this has been added here for simplicity. Also, keep in mind that, in `prodRelease` build type, R8 will probably strip these classes away.

**Reply by JackEblan** [reply]
Is it okay to move these to demo flavour including this https://github.com/android/nowinandroid/blob/ab9033421017962c5af6d7a1dd221997439210da/core/network/src/main/kotlin/com/google/samples/apps/nowinandroid/core/network/di/NetworkModule.kt#L49

**Reply by SimonMarquis** [reply]
I think you'll also need to move *all* the test classes that rely on it, and this is probably why it remained here 😕

**Reply by JackEblan** [reply]
Hehe sorry😁 I don't want to touch it no more.
